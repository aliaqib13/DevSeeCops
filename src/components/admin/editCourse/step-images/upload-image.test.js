import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Resizer from 'react-image-file-resizer';
import UploadImage from './upload-image';

const state = {
    imageUrl: 'data:image/png;base64,testimageurl',
    height: 66,
    width: 72,
    heightRatio: '0.9',
    widthRatio: '1.1',
    file: {
        uid: 'rc-upload-1591961028788-2',
    },
};

describe('Upload Image', () => {
    let component;

    beforeEach(() => {
        component = shallow(<UploadImage />);
    });

    it('Should render successfully', () => {
        const wrapper = component.find('.upload-image');
        expect(wrapper.exists()).toBeTruthy();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render buttons disables when there is no image uploaded', () => {
        const actionButtons = component.find('.actions').find('Button');
        expect(actionButtons.at(0).props().disabled).toBeTruthy();
        expect(actionButtons.at(1).props().disabled).toBeTruthy();
        expect(actionButtons.at(2).props().disabled).toBeTruthy();
    });

    it('Should render inputs for changing size disabled when there is no image uploaded', () => {
        const inputs = component.find('.resize-inputs').find('InputNumber');
        expect(inputs.at(0).props().disabled).toBeTruthy();
        expect(inputs.at(1).props().disabled).toBeTruthy();
    });
});

describe('Upload Image component with uploaded image', () => {
    let component;

    beforeEach(() => {
        component = shallow(<UploadImage />);
        component.instance().setState(state);
    });

    it('Should add image to upload field when image is uploaded', () => {
        expect(component.find('Upload').find('img').props().src).toBe(state.imageUrl);
    });

    it('Should show modal with image when clicked on preview button', () => {
        const previewButton = component.find('.actions Button').at(0);
        previewButton.simulate('click');
        const modal = component.find('Modal');
        expect(modal.props().visible).toBeTruthy();
    });

    it('Should render modal with proper image', () => {
        const previewButton = component.find('.actions').find('Button').at(0);
        previewButton.simulate('click');
        const modal = component.find('Modal');
        expect(modal.find('img').props().src).toBe(state.imageUrl);
    });

    it('Should call uploadImage method when clicked on "Resize and Upload" button', () => {
        Resizer.imageFileResizer = jest.fn();
        const spy = jest.spyOn(component.instance(), 'uploadImage');
        component.instance().setState(state);
        const ResizeAndSave = component.find('.actions Button').at(1);
        ResizeAndSave.simulate('click');
        expect(spy).toBeCalledTimes(1);
    });

    it('Should render modal title with right dimensions', () => {
        const modal = component.find('Modal');
        expect(modal.props().title).toBe(`Image Preview: ${state.height} x ${state.width}`);
    });
});
