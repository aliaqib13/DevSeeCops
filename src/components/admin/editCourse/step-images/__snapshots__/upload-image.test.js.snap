// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Image Should render successfully 1`] = `
<div
  className="upload-image"
>
  <Upload
    accept=""
    action=""
    beforeUpload={[Function]}
    className="avatar-uploader"
    customRequest={[Function]}
    data={Object {}}
    disabled={false}
    listType="picture-card"
    multiple={false}
    name="step"
    onChange={[Function]}
    showUploadList={false}
    supportServerRender={true}
    type="select"
  >
    <div>
      <Icon
        type="plus"
      />
      <div
        className="ant-upload-text"
      >
        Upload Image
      </div>
    </div>
  </Upload>
  <div
    className="resize-inputs"
  >
    <label>
      Height
    </label>
    <InputNumber
      disabled={true}
      max={5000}
      min={1}
      onChange={[Function]}
      size="small"
      step={1}
      value={1}
    />
    <label>
      Width
    </label>
    <InputNumber
      disabled={true}
      max={5000}
      min={1}
      onChange={[Function]}
      size="small"
      step={1}
      value={1}
    />
  </div>
  <div
    className="actions"
  >
    <Button
      block={false}
      disabled={true}
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
      size="small"
    >
      Preview
    </Button>
    <Button
      block={false}
      disabled={true}
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
      size="small"
      type="primary"
    >
      Resize and Upload
    </Button>
    <Button
      block={false}
      disabled={true}
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
      size="small"
      type="danger"
    >
      Clear
    </Button>
  </div>
  <Modal
    cancelText="Close"
    confirmLoading={false}
    maskTransitionName="fade"
    okText="Resize and Upload"
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    title="Image Preview: 1 x 1"
    transitionName="zoom"
    visible={false}
    width="70%"
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <img
        alt="step"
        height={1}
        src=""
        style={
          Object {
            "maxWidth": "100%",
          }
        }
        width={1}
      />
    </div>
  </Modal>
</div>
`;
